src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -95,13 +95,35 @@ internal partial class CSharpIntroduceVariableService
             var rewrittenBody = Rewrite(
                 document, expression, newLocalName, document, oldBody, allOccurrences, cancellationToken);
 
+            var doesDelegateMethodReturnVoid =
+                document.SemanticModel.GetTypeInfo(oldLambda, cancellationToken).ConvertedType is INamedTypeSymbol delegateType &&
+                delegateType.DelegateInvokeMethod != null &&
+                delegateType.DelegateInvokeMethod.ReturnsVoid;
 
-            var newBody =
-                document.SemanticModel.GetTypeInfo(oldLambda, cancellationToken).ConvertedType is INamedTypeSymbol delegateType
-                && delegateType.DelegateInvokeMethod != null
-                && delegateType.DelegateInvokeMethod.ReturnsVoid
-                    ? SyntaxFactory.Block(declarationStatement)
-                    : SyntaxFactory.Block(declarationStatement, SyntaxFactory.ReturnStatement(rewrittenBody));
+            BlockSyntax newBody;
+            if (doesDelegateMethodReturnVoid)
+            {
+                // We don't need to include the rewritten body if it consists of just the identifier name.
+                // However, in all other cases, the rewritten body should be included.
+                var isRewrittenBodyTrivial = rewrittenBody.WalkDownParentheses().IsKind(SyntaxKind.IdentifierName,
+                    out IdentifierNameSyntax identifier) &&
+                    identifier.Identifier.ValueText == newLocalName.Identifier.ValueText;
+
+                if (isRewrittenBodyTrivial)
+                {
+                    newBody = SyntaxFactory.Block(declarationStatement);
+                }
+                else
+                {
+                    newBody = SyntaxFactory.Block(
+                        declarationStatement,
+                        SyntaxFactory.ExpressionStatement(rewrittenBody, SyntaxFactory.Token(SyntaxKind.SemicolonToken)));
+                }
+            }
+            else
+            {
+                newBody = SyntaxFactory.Block(declarationStatement, SyntaxFactory.ReturnStatement(rewrittenBody));
+            }
```

Comment:
i basically don't understand most of what's going on here.  specifically why this is correct and doing the right thing for all the cases that might come through here

Reply:
The non-void return case existed in the previous code for cases like:
```C# 
        Func<int, int> f = x => [|x + 1|];
```

```C#
        Func<int, int> f = x =>
        {
            var v = x + 1;
            return v;
        }
```